jerry-core/parser/regexp/re-compiler.c

```diff
@@ -445,23 +445,23 @@ re_parse_alternative (re_compiler_ctx_t *re_ctx_p, /**< RegExp compiler context
 } /* re_parse_alternative */
 
 static const re_compiled_code_t *re_cache[RE_CACHE_SIZE];
+static uint16_t re_cache_idx = RE_CACHE_SIZE;
 
 /**
  * Search for the given pattern in the RegExp cache
  *
- * @return compiled bytecode - if found
- *         NULL              - otherwise
+ * @return index of bytecode in cache - if found
+ *         RE_CACHE_SIZE              - otherwise
  */
-const re_compiled_code_t *
+static uint16_t
```

Comment:
I see now that re_cache_idx is uint16_t. If we want to force a small number we could use uint8_t. Anyway I am even happy with uint32_t as well.
Otherwise LGTM.